libs/httpext/src/lib/plugin.spec.ts

```diff
@@ -1,23 +1,26 @@
-import { TestBed, async, fakeAsync } from '@angular/core/testing';
-import {
-  HttpClientTestingModule,
-  HttpTestingController
-} from '@angular/common/http/testing';
-import {
-  HttpClient,
-  HTTP_INTERCEPTORS,
-  HttpClientModule
-} from '@angular/common/http';
+import { HttpClient } from '@angular/common/http';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { TestBed } from '@angular/core/testing';
+
 import { HttpExtModule } from './http-ext.module';
-import { useCachePlugin } from './cache-plugin';
+import { Plugin } from './plugin';
+
+function loggerPlugin(): Plugin {
+  return {
+    handle({ req, next }) {
+      console.log('hello world');
```

Comment:
🤔 what about creating the `loggerPlugin` inside the describe and call a spy explicitly instead of `console.log` and then mocking `console.log`? I'd rather avoid mocking global functions when possible specially `console.log`.

Reply:
✅ created a dedicated mock and checking the request is passed to the plugin handler.
looks good for you?